City-of-Helsinki/helsinki-paatokset
# UHF-10527: Fetch meetings from ElasticSearch

# [UHF-10527](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10527)

## What was done
- Meeting list data is now fetched from elasticsearch
- Old database queries have been removed

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-10527-meetings-es-implementation`
* Recommended to use a production DB dump for more up-to-date meeting and policymaker content.
  * Alternatively, you can use the test content module: `drush en -y paatokset_test_content`
* Fetch relevant data (all policymakers and meetings for the last few years)
  * `drush mim ahjo_decisionmakers:all --update;drush mim ahjo_decisionmakers:all_sv --update`
  * `drush ap:agg meetings --start=2022-01-01T00:00:00 --end=2022-12-31T00:00:00 --queue -v;drush ap:agg meetings --start=2023-01-01T00:00:00 --end=2023-12-31T00:00:00 --queue -v;drush ap:agg meetings --start=2024-01-01T00:00:00 --end=2024-12-31T00:00:00 --queue -v; drush queue:run ahjo_api_aggregation_queue`
* Index all meetings content with: `drush sapi-c meetings;drush sapi-rt meetings;drush sapi-i meetings;`

## How to test
* Check that this feature works. Compare each of these pages to the versions in production, they should have the same data
  * [x] Frontpage calendar block: https://helsinki-paatokset.docker.so/fi
  * [x] Meeting calendar: https://helsinki-paatokset.docker.so/fi/kokouskalenteri
  * Organization pages (check the organization main page's document and upcoming meetings lists and then the documents listing page)
    * [x] https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginvaltuusto
    * [x] https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginhallitus
    * [x] Also pick a few random organizations (not office holders) from https://helsinki-paatokset.docker.so/fi/paattajat/ that have meetings
* Test caching functionality
  * [x] Make sure you have page caching turned on (comment out the cache disabling lines from local.settings.php) 
  * [x] Find an upcoming meeting that is visible on the frontpage, on an organization's page and in the meeting calendar. Then edit the meeting node (https://helsinki-paatokset.docker.so/fi/kokoukset/[meeting-id]) and change the displayed meeting time by a few hours
  * [x] Run `drush sapi-i meetings`
  * [x] Reload all the pages where the meeting was visible. The time should have changed (and at least on the meeting calendar it should have a note about having been moved)
  * [x] Find another meeting, this time on the organization's document list. Do the same operation with editing the time and then reindex meetings. The changes should appear on the document list and on the document page after reindexing.

## Continuous documentation
* [x] This feature has been documented/the documentation has been updated: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/PP/pages/8333819923/Kriittiset+toiminnallisuudet+ja+ongelmatilanteiden+selvitys#Kokousdata-ja-toimielinten-asiakirjat

[UHF-10527]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
